### ISSUE TITLE
Android: onDateChange callback in not being called for `type` time

### ISSUE THREAD

**Issue by olegmilan** [henninghall/react-native-date-picker#111]
**Bug description**
Currently `onDateChange` callback is being called only for `date` and `datetime`.
But for `time` nothing happens and callback not being called

See code:
```javascript
handleDatePicked = (date) => {
    // some code
    log('handleDatePicked date changed', date);
};

render() {
    return (
        <DatePicker
          date={new Date()}
          mode={'time'}
          onDateChange={this.handleDatePicked}
          locale={'en_GB'}
        />
    );
  }
```

**Expected behavior**
`onDateChange` to be called for each mode `time`, `date`, `datetime`

**Android emulator - Nexus_5X_API_28_x86 :**
 - OS: [Android]
 - React Native version [e.g. 0.60.4]
 - react-native-date-picker version [e.g. 2.7.0]

**Comment by henninghall** [comment]
Thanks for reporting this, I will have a look at it!

**Comment by olegmilan** [comment]
> Thanks for reporting this, I will have a look at it!

After investigation, I found that `onDateChange` is not being called when `locale` param is specified when mode: `time`

**Comment by Humbl-e** [comment]
I have the same issue. I did some digging but when mode is time and locale is anything other than "en" or "en_US" , the ONCHANGE listener throws an exception where the dateformat can not be parsed, hence there is no date returned onDateChange.
hope it helps.

**Comment by lgalant** [comment]
Found the same issue with my Samsung s9. However it works fine on my emulator (Nexus 5X 28)

**Comment by tzwickl** [comment]
I found the same issue when setting a locale. Anybody here who can fix this?

**Comment by henninghall** [comment]
@tzwickl I don't have had any time to look into this, sorry! I will have a look at this and the other issues after new years. In the meantime, you can try to see if locale `en`, `en_US` or another locale will give you a result that works for you.

**Comment by olegmilan** [comment]
Hey @tzwickl @lgalant @Humbl-e I managed to fix this by forking this repo and adding early `return false` to the function `usesAmPm` in `/node_modules/react-native-date-picker/android/src/main/java/com/henninghall/date_picker/Settings.java`:

```
    public static boolean usesAmPm (){
    	// We always want 24 Hours format
    	return false;
        // return !DateFormat.is24HourFormat(DatePickerManager.context);
    }
```

It is not ideal, but we like this library and it's performance, so we decided to go with this hack

**Comment by janwasgint** [comment]
@henninghall I also encountered this issue when setting the locale on Android to arabic and mode to `datetime`. It would be really great to have that work. What's the current state on it?

**Comment by henninghall** [comment]
@janwasgint It definitely has high priority and I will hopefully have the time to fix it soon!

**Comment by ezranbayantemur** [comment]
Same issue +1

**Comment by henninghall** [comment]
Released in v2.7.4

**Comment by henninghall** [comment]
Fixed and released in v2.7.6
